apps/internal/base/base.go

```diff
@@ -345,6 +349,26 @@ func (b Client) AcquireTokenSilent(ctx context.Context, silent AcquireTokenSilen
 	if silent.Claims == "" {
 		ar, err = AuthResultFromStorage(storageTokenResponse)
 		if err == nil {
+			if shouldRefresh(storageTokenResponse.AccessToken.RefreshOn.T) {
+				if tr, er := b.Token.Credential(ctx, authParams, silent.Credential); er == nil {
```

Comment:
This assumes the caller is a confidential client, but actually public clients call this method as well. Because they pass `silent.Credential == nil`, they panic when the cached token should be refreshed.